shared/actions/settings.js

```diff
@@ -77,6 +89,24 @@ function * _onUpdatePGPSettings (): SagaGenerator<any, any> {
   }
 }
 
+function * _onSubmitNewEmail (): SagaGenerator<any, any> {
+  try {
+    const selector = (state: TypedState) => state.settings.email
```

Comment:
No, this is worse since the selector fn `state => state.settings.email.newEmail` won't have any type safety